### PR TITLE
Adding timeout param with default timeout. 

### PR DESCRIPTION
**Pull request by soumyasmruti**

### CODE REVIEW

File: docker/wait-for-db.sh [hunk 1]
```diff
@@ -12,21 +12,23 @@
...
-  sleep 1
-done
+        --command '"'"'\q'"'"' > /dev/null 2>&1; do 
+  echo "Waiting for postgres to become available...";
+  sleep 1;
+done;
+echo "Great news! Postgres is up."';
 
-echo "Great news! Postgres is up."
+echo "Timeout, the postgres server is taking too long to respond."
```
Comment (ashulmanWeWork) [inline docker/wait-for-db.sh]: What prevents line 34 from being called every time?

Comment (soumyasmruti) [reply docker/wait-for-db.sh]: Added the message after the or statement.

File: docker/wait-for-db.sh [hunk 2]
```diff
@@ -25,10 +25,8 @@ timeout time bash -c 'until PGPASSWORD="${POSTGRES_PASSWORD}" psql \
...
         --dbname "${POSTGRES_DB}" \
-        --command '"'"'\q'"'"' > /dev/null 2>&1; do 
+        --command '"'"'\q'"'"' > /dev/null 2>&1; do
   echo "Waiting for postgres to become available...";
   sleep 1;
 done;
-echo "Great news! Postgres is up."';
-
-echo "Timeout, the postgres server is taking too long to respond."
+echo "Great news! Postgres is up."' || echo "Timeout, the postgres server is taking too long to respond." ;
```
Comment (ashulmanWeWork) [inline docker/wait-for-db.sh]: I'm a little confused by this line. What is it supposed to output the first output if the script is successful and the second one if it's unsuccessful? If so, I'm not sure how the outcome of the connection to the database would affect this line.

File: docker/wait-for-db.sh [hunk 3]
```diff
@@ -12,21 +12,21 @@
...
 # limitations under the License.
 #
-# Usage: $ ./wait-for-db.sh <host> <port>
+# Usage: $ ./wait-for-db.sh <host> <port> [timeout]
 
 set -eu
 
 host="${1}"
 port="${2}"
+time="${3}" || 60s
```
Comment (wslulciuc) [inline docker/wait-for-db.sh]: _Minor_: Let's default the `time` param as:

```
time="${3:-60s}"
```

**Comment by wslulciuc** [comment]
@soumyasmruti, should I review this PR?

File: docker/wait-for-db.sh [hunk 4]
```diff
@@ -12,21 +12,21 @@
...
 
-until PGPASSWORD="${POSTGRES_PASSWORD}" psql \
+timeout time bash -c 'until PGPASSWORD="${POSTGRES_PASSWORD}" psql \
```
Comment (wslulciuc) [inline docker/wait-for-db.sh]: `time` is a reserved keyword in bash

File: docker/wait-for-db.sh (hunk 2 above)
Comment (soumyasmruti) [reply docker/wait-for-db.sh]: added if else statement to make it more clear.

File: docker/wait-for-db.sh (hunk 3 above)
Comment (soumyasmruti) [reply docker/wait-for-db.sh]: Done.

File: docker/wait-for-db.sh (hunk 4 above)
Comment (soumyasmruti) [reply docker/wait-for-db.sh]: changed.

**Comment by soumyasmruti** [comment]
@wslulciuc Yes I have completed my testing, did few changes. This is ready to be reviewed.

File: docker/wait-for-db.sh [hunk 5]
```diff
@@ -12,21 +12,16 @@
...
 # limitations under the License.
 #
-# Usage: $ ./wait-for-db.sh <host> <port>
+# Usage: $ ./wait-for-db.sh <host> <port> [timeout]
 
 set -eu
 
 host="${1}"
 port="${2}"
+timeInSecs="${3:-60s}"
```
[the remark this answers is not in the document]
Comment (soumyasmruti) [reply docker/wait-for-db.sh]: Done

**Comment by wslulciuc** [comment]
@soumyasmruti, I decided in going with @ajaymuppuri solution for adding a `timeout` flag to `wait-for-db.sh` (mainly for the simpler logic). Thanks again for exploring an alternative approach!

**Pull request closed without merging**